### PR TITLE
Adds Checkout 

### PR DESCRIPTION
**Pull request by tlgimenes**
This PR adds checkout capabilities to the store. The main ideia is to add redirects to `/checkout` and make the cookie availiable to legacy checkout APIs.

Also, this PR removed unecessary types and fetch-related code. We could remove it because we were using the API directly instead of using the graphql API

[marinbrasil](https://github.com/vtex-sites/marinbrasil.store/pull/98)
[faststore](https://github.com/vtex-sites/storecomponents.store/pull/293)

### CODE REVIEW

File: packages/gatsby-theme-vtex/src/sdk/graphql/request.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import {
+  request as baseRequest,
+  RequestOptions as GraphQLRequestOptions,
+} from '@vtex/gatsby-plugin-graphql'
+
+export type RequestOptions = GraphQLRequestOptions
+
+export const request = async <
+  Query extends any = any,
+  Variables extends any = any
```
Comment (emersonlaurentino) [inline packages/gatsby-theme-vtex/src/sdk/graphql/request.ts:10]: why don't you use any and any?

File: packages/gatsby-theme-vtex/src/sdk/orderForm/controller/orderForm.ts [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+
+export const startOrderForm = async (
+  orderForm: OrderFormFragment_OrderFormFragment,
+  cb: CB
+) => {
+  setOrderFormState(orderForm, cb)
+  queue.start()
+}
+
+export const fragment = gql`
```
Comment (emersonlaurentino) [inline packages/gatsby-theme-vtex/src/sdk/orderForm/controller/orderForm.ts:34]: HUGE

**Review by emersonlaurentino — APPROVED** [review]

**Review by lucis — APPROVED** [review]

File: packages/gatsby-theme-vtex/src/sdk/orderForm/controller/orderForm.ts (hunk 2 above)
Comment (tlgimenes) [reply packages/gatsby-theme-vtex/src/sdk/orderForm/controller/orderForm.ts:34]: copied from:
https://github.com/vtex-apps/checkout-resources/blob/3db2f672d20ab59502e859da683c0fea226f056b/react/fragments/orderForm.graphql#L4

File: packages/gatsby-theme-vtex/src/sdk/graphql/request.ts (hunk 1 above)
Comment (tlgimenes) [reply packages/gatsby-theme-vtex/src/sdk/graphql/request.ts:10]: This was copied from [here](https://github.com/vtex/faststore/blob/ea05570a1a7d5472b4eb557e3be2851569415168/packages/gatsby-plugin-graphql/src/index.ts#L20).

**Pull request merged**